### PR TITLE
Support package refactor

### PR DESCRIPTION
**Pull request by Fiona-Waters**
# Issue link
https://github.com/project-codeflare/codeflare-operator/issues/237

# What changes have been made
- Moved the support files from the Codeflare Operator Repo to this new separate repo for common functions.
- Added Makefile and moved `verify-imports.sh` in order to be able to verify imports are correct.

Questions:
- Do we want to group the files further within the support directory?
- We should have an OWNERS file - Who should be added as approvers and reviewers?
- Do we want to add PR checks for this repo now?

# Verification Steps
- Once the [Codeflare Operator repo](https://github.com/project-codeflare/codeflare-operator/pull/336) is updated to reflect these changes we can verify that everything works as expected.

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Review by ChristianZaccaria — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+## Location to install dependencies to
+LOCALBIN ?= $(shell pwd)/bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
```
Comment (astefanutti) [inline Makefile:5]: Should we copy the target that overwrites values in `support/defaults.go`?

Comment (Fiona-Waters) [reply Makefile:5]: Thank you @astefanutti , yes I can move that over. Is it okay to remove that make target from CFO entirely? From a look at the use of the values in defaults.go having them just in the new repo should suffice.

Comment (astefanutti) [reply Makefile:38]: Yes I think so too. There are also environment variables that can be used to overwrite these defaults.

Comment (Fiona-Waters) [reply Makefile:38]: Ok I will remove it then. Thank you. 
I also need some input on reviewers/approvers for the repo - will I copy these over from CFO as they are there?
And should I add any PR checks/actions at this point?

Comment (astefanutti) [reply Makefile:38]: Yes we can do that in a separate PR if you want. We'll also have to setup openshift-ci for the repo.

Comment (Fiona-Waters) [reply Makefile:38]: Ok I will create a follow up issue for these. Thank you!

Comment (Fiona-Waters) [reply Makefile:38]: Follow on issue: https://github.com/project-codeflare/codeflare-common/issues/2 
I've added the `needs-triage` label but have not added it to any sprint.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+## Location to install dependencies to
+LOCALBIN ?= $(shell pwd)/bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+OPENSHIFT-GOIMPORTS ?= $(LOCALBIN)/openshift-goimports
+
+.PHONY: openshift-goimports
+openshift-goimports: $(OPENSHIFT-GOIMPORTS) ## Download openshift-goimports locally if necessary.
```
[the remark this answers is not in the document]
Comment (Fiona-Waters) [reply Makefile:42]: `verify_generated_files` workflow added.

**Review by astefanutti — APPROVED** [review]

**Pull request merged**